terraform/modules/iot-edge/main.tf

```diff
@@ -31,15 +31,27 @@ resource "azurerm_public_ip" "iot_edge" {
 }
 
 resource "azurerm_network_security_group" "iot_edge" {
-  name                = "nsg-${local.dns_label_prefix}"
+  name                = "nsg-iotedge-${var.resource_uid}"
   resource_group_name = var.resource_group_name
   location            = var.location
 
   security_rule {
-    name                       = "nsgr-allow-22"
-    priority                   = 1000
+    name                       = "AllowBastionInbound"
+    priority                   = 900
     access                     = "Allow"
     direction                  = "Inbound"
+    protocol                   = "*"
+    source_address_prefix      = "10.0.2.0/26"
+    source_port_range          = "*"
+    destination_address_prefix = "*"
+    destination_port_ranges    = ["3389", "22"]
+  }
+
+  security_rule {
+    name                       = "Default-Deny-22"
```

Comment:
Let's use the CAF recommended abbreviation `nsgsr-` (I incorrectly put it as `nsgr` before)